src/utils.jl

```diff
@@ -20,3 +24,20 @@ ShiftedProximalOperators.iprox!(
 
 LinearAlgebra.diag(op::AbstractDiagonalQuasiNewtonOperator) = copy(op.d)
 LinearAlgebra.diag(op::SpectralGradient{T}) where {T} = zeros(T, op.nrow) .* op.d[1]
+
+"""
+    GenericExecutionStats(reg_nlp :: AbstractRegularizedNLPModel{T, V})
+
+Construct a GenericExecutionStats object from an AbstractRegularizedNLPModel. 
+More specifically, construct a GenericExecutionStats on the NLPModel of reg_nlp and add three solver_specific entries namely :smooth_obj, :nonsmooth_obj and :xi.
+This is useful for reducing the number of allocations when calling solve!(..., reg_nlp, stats) and should be used by default.
+Warning: This should *not* be used when adding other solver_specific entries that do not have the current scalar type. 
+For instance, when one adds the history of the objective value as a solver_specific entry (which has Vector{T} type), this will cause an error and `GenericExecutionStats(reg_nlp.model)` should be used instead.
+"""
+function GenericExecutionStats(reg_nlp :: AbstractRegularizedNLPModel{T, V}) where{T, V}
+  stats = GenericExecutionStats(reg_nlp.model, solver_specific = Dict{Symbol, T}())
+  set_solver_specific!(stats, :smooth_obj, T(Inf))
+  set_solver_specific!(stats, :nonsmooth_obj, T(Inf))
+  set_solver_specific!(stats, :xi, T(Inf))
```

Comment:
Wouldn’t $\xi$ be in `dual_feas`?

Reply:
I don't see how $$\xi$$ is related with dual feasibility of the unconstrained problem but perhaps I am missing something ?